samples/AzureSample.ResourceManager.Sample/src/Generated/ImageData.Serialization.cs

```diff
@@ -362,13 +357,18 @@ private BinaryData SerializeBicep(ModelReaderWriterOptions options)
 
             builder.Append("  properties:");
             builder.AppendLine(" {");
-            hasPropertyOverride = hasObjectOverride && propertyOverrides.TryGetValue(nameof(SourceVirtualMachine), out propertyOverride);
+            hasPropertyOverride = hasObjectOverride && propertyOverrides.TryGetValue("SourceVirtualMachineId", out propertyOverride);
             if (Optional.IsDefined(SourceVirtualMachine) || hasPropertyOverride)
             {
                 builder.Append("    sourceVirtualMachine: ");
                 if (hasPropertyOverride)
                 {
-                    builder.AppendLine($"{propertyOverride}");
+                    builder.AppendLine("{");
```

Comment:
When the property gets a second value, it will become unflattened. The user's override would no longer work, but it would be difficult to handle this scenario because the user could override the entire envelope when the property is unflattened. We would need to come up with an algorithm for determining precedence. I would suggest that we punt this for now given that we can address it a non-breaking way if needed in the future.